### ISSUE TITLE
Command interface

### ISSUE THREAD

**Issue by civilframe** [civilframe/insurrection#1]
Accept user commands, act on the game world, display results.

**Comment by jbauschatz** [comment]
A Command might have a "keyword" that uniquely identifies it. It might also accept any number of arguments. I'd picture something like this:

``` Java
interface Command {
     public String getKeyword();
     public void execute(String[] args, Game game);
}
```
So maybe a "Research" command could be executed by typing "Research WarpDrive" - where "Research" is the keyword that identifies the command, and "WarpDrive" is passed in the args.

**Comment by civilframe** [comment]
Yeah, that's spot on. I like the functional style of passing the game in.
